modules/nats/nats_test.go

```diff
@@ -57,3 +59,45 @@ func TestNATS(t *testing.T) {
 
 	require.Equal(t, "hello", string(msg.Data))
 }
+
+func TestNATSWithConfigFile(t *testing.T) {
+	const natsConf = `
+listen: 0.0.0.0:4222
+authorization {
+    token: "s3cr3t"
+}
+`
+	ctx := context.Background()
+
+	ctr, err := tcnats.Run(ctx, "nats:2.9", tcnats.WithConfigFile(strings.NewReader(natsConf)))
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	uri, err := ctr.ConnectionString(ctx)
+	require.NoError(t, err)
+
+	// connect without a correct token must fail
+	mallory, err := nats.Connect(uri, nats.Name("Mallory"), nats.Token("secret"))
+	require.Error(t, err)
+	require.ErrorIs(t, err, nats.ErrAuthorization)
+	t.Cleanup(mallory.Close)
+
+	// connect with a correct token must succeed
+	nc, err := nats.Connect(uri, nats.Name("API Token Test"), nats.Token("s3cr3t"))
+	require.NoError(t, err)
+	t.Cleanup(nc.Close)
+
+	// validate /etc/nats.conf mentioned in logs
+	const expected = "Using configuration file: /etc/nats.conf"
+	logs, err := ctr.Logs(ctx)
+	require.NoError(t, err)
+	sc := bufio.NewScanner(logs)
```

Comment:
A
suggestion: this looks like it will deadlock if the string is not found, add a timeout to avoid that.

Reply:
used `time.After` in ff17ba124cb6cf9c72bfd04206630fd4afb6fb8a